src/main/java/com/epam/aidial/core/util/BufferingReadStream.java

```diff
@@ -24,7 +27,7 @@ public class BufferingReadStream implements ReadStream<Buffer> {
     private boolean ended;
     private boolean reset;
     private final boolean isStreaming;
-    private Buffer lastChunk;
+    private List<Buffer> lastChunks;
```

Comment:
just remember position and make a slice from content buffer

Reply:
fixed